[Cocoa] Temporarily disable accessibility bold tests while it's being reimplemented

litherum opened the pull request:
#### 5790c03c39744699de52d4e8b385b4baf1e18e90
<pre>
[Cocoa] Temporarily disable accessibility bold tests while it&apos;s being reimplemented
<a href="https://bugs.webkit.org/show_bug.cgi?id=242023">https://bugs.webkit.org/show_bug.cgi?id=242023</a>

Reviewed by Simon Fraser.

This is a temporary patch to disable the accessibility bold tests while
<a href="https://bugs.webkit.org/show_bug.cgi?id=237817">https://bugs.webkit.org/show_bug.cgi?id=237817</a> is being worked on. Because
the mechanism for accessibility bold is changing (in that bug), the
testing mechanism will need to be rewritten. In the interest of making this
stuff reviewable, this is a precursor patch to the reimplementation.

* LayoutTests/TestExpectations:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* Source/WebCore/platform/graphics/FontCache.h:
(WebCore::FontCache::shouldMockBoldSystemFontForAccessibility const): Deleted.
(WebCore::FontCache::setShouldMockBoldSystemFontForAccessibility): Deleted.
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::fontWithFamilySpecialCase):
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::systemFontParameters):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::shouldMockBoldSystemFontForAccessibility const): Deleted.
(WebCore::RenderTheme::setShouldMockBoldSystemFontForAccessibility): Deleted.
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::Backup::Backup):
(WebCore::InternalSettings::Backup::restoreTo):
(WebCore::InternalSettings::setShouldMockBoldSystemFontForAccessibility):
* Source/WebCore/testing/InternalSettings.h:

Canonical link: <a href="https://commits.webkit.org/251874@main">https://commits.webkit.org/251874@main</a>
</pre>

smfr commented on LayoutTests/TestExpectations:
```diff
@@ -5320,3 +5317,7 @@ fast/attachment/attachment-image-controls-basic.html [ Skip ]
 fast/text/bulgarian-system-language-shaping.html [ ImageOnlyFailure ]
+
+# Temporarily disable accessibility bold tests while it's being reimplemented.
+fast/text/accessibility-bold-system-font.html [ Pass Failure ]
```
Is there a bugzilla bug you can reference here?

smfr commented on Source/WebCore/testing/InternalSettings.cpp:
```diff
@@ -572,9 +567,8 @@ ExceptionOr<void> InternalSettings::setShouldMockBoldSystemFontForAccessibility(
-    // FIXME: Call setShouldMockBoldSystemFontForAccessibility() on all workers.
-    FontCache::forCurrentThread().setShouldMockBoldSystemFontForAccessibility(should);
+    UNUSED_PARAM(should);
+    // FIXME: Implement this.
```
Maybe the FIXME could have a little more context (like a bug URL).

litherum committed:
[Cocoa] Temporarily disable accessibility bold tests while it's being reimplemented
https://bugs.webkit.org/show_bug.cgi?id=242023

Reviewed by Simon Fraser.

This is a temporary patch to disable the accessibility bold tests while
https://bugs.webkit.org/show_bug.cgi?id=237817 is being worked on. Because
the mechanism for accessibility bold is changing (in that bug), the
testing mechanism will need to be rewritten. In the interest of making this
stuff reviewable, this is a precursor patch to the reimplementation.

* LayoutTests/TestExpectations:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* Source/WebCore/platform/graphics/FontCache.h:
(WebCore::FontCache::shouldMockBoldSystemFontForAccessibility const): Deleted.
(WebCore::FontCache::setShouldMockBoldSystemFontForAccessibility): Deleted.
* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::fontWithFamilySpecialCase):
* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::systemFontParameters):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::shouldMockBoldSystemFontForAccessibility const): Deleted.
(WebCore::RenderTheme::setShouldMockBoldSystemFontForAccessibility): Deleted.
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::Backup::Backup):
(WebCore::InternalSettings::Backup::restoreTo):
(WebCore::InternalSettings::setShouldMockBoldSystemFontForAccessibility):
* Source/WebCore/testing/InternalSettings.h:

Canonical link: https://commits.webkit.org/251874@main
